client/components/ReviewRating/WriteReview/WriteReview.jsx

```diff
@@ -0,0 +1,195 @@
+  };
+  const saveUrl = () => {
+    if (newReview.photos && (newReview.photos.match(/\.(jpeg|jpg|gif|png)$/) != null)) {
+      setPhotoUrl([...photoUrl, newReview.photos]);
+    } else {
+      document.getElementById('photos').classList.add('error');
+      document.getElementById('photos').placeholder = 'invalid ulr';
+    }
+    setNewReview({ ...newReview, photos: '' });
+  };
+  const onChangeHandler = (e) => {
+    if (e.target.name !== 'recommend') {
+      document.getElementById(e.target.name).classList.remove('error');
+    }
+    if (charName.includes(e.target.name)) {
+      // eslint-disable-next-line max-len
+      const copyCharValue = [...charValue];
+      copyCharValue[charName.indexOf(e.target.name)] = e.target.value;
+      setCharValue(copyCharValue);
+    } else {
+      setNewReview({ ...newReview, [e.target.name]: e.target.value });
+    }
+  };
+
+  const onSubmit = (e) => {
+    setIsValid(true);
+    e.preventDefault();
+    if (!newReview.summary) {
+      document.getElementById('summary').classList.add('error');
+      document.getElementById('summary').placeholder = 'can not be empty';
+      setIsValid(false);
+    }
+    if (!newReview.body) {
+      document.getElementById('body').classList.add('error');
+      document.getElementById('body').placeholder = 'can not be empty';
+      setIsValid(false);
+    }
+    if (!newReview.email) {
+      document.getElementById('email').classList.add('error');
+      document.getElementById('email').placeholder = 'can not be empty';
+      setIsValid(false);
+    }
+    if (!newReview.name) {
+      document.getElementById('name').classList.add('error');
+      document.getElementById('name').placeholder = 'can not be empty';
+      setIsValid(false);
+    }
+    if (newReview.email && !looksLikeMail(newReview.email)) {
+      document.getElementById('email').classList.add('error');
+      setNewReview({ ...newReview, email: '' });
+      document.getElementById('email').placeholder = 'invalid email';
+      setIsValid(false);
+    }
+    if (newReview.body && newReview.body.length < 5) {
+      document.getElementById('body').classList.add('error');
+      setNewReview({ ...newReview, body: '' });
+      document.getElementById('body').placeholder = 'must be greater than 50 characters';
+      setIsValid(false);
+    }
+    if (isValid) {
+      const review = newReview;
+      review.photos = photoUrl;
+      review.rating = Number(review.rating);
+      for (let i = 0; i < charName.length; i += 1) {
+        review.characteristics[charId[i]] = Number(charValue[i]);
+      }
+
+      if (review.name !== '') {
+        axios.post('/reviews', review);
+        closeWriteReview();
+      }
+    }
+  };
+  return (
+    <div className={isClickAdd ? 'window' : 'close'}>
+      <form className="modal">
+        <h3>Review</h3>
+        <div className="inputLabel">
```

Comment:
Maybe create a TODO item to clean this file up. There seems to be quite a bit of repeated code.

Reply:
absolutely, I was trying to get the function to work. Ticket added.